File: Python/Easy/Caesar Chipher Encryptor/solution_1.py
def caesarCipherEncryptor(string, key):
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    input_idx = 0
    lst = []

    while input_idx < len(string):

        letterInString = string[input_idx]
        #In order to make it more abstract
        nextLetter = findAndShift(letterInString, key, alphabet)
        lst.append(nextLetter)
        input_idx += 1
    
    return "".join(lst)


def findAndShift(letter, shift, alphabet) -> str:

    new_letterIdx = alphabet.find(letter) + shift
    return alphabet[new_letterIdx % 26]


    
    

File: Go/Easy/Caesar Cipher Encryptor/solution_1.go
package main

import "strings"

func CaesarCipherEncryptor(str string, key int) string {

	alphabet := "abcdefghijklmnopqrstuvwxyz"
	var lst []rune
	input_idx := 0

	for input_idx < len(str) {

		firstLetter := rune(str[input_idx])
		//Abstraction
		newLetter := findAndShift(firstLetter, key, alphabet)
		lst = append(lst, newLetter)
		input_idx += 1

	}

	return string(lst)
}

func findAndShift(letter rune, shift int, alphabet string) rune {

	newLetterCode := (strings.Index(alphabet, string(letter)) + shift) % 26
	newLetter := alphabet[newLetterCode]
	return rune(newLetter)

}
